frontend/src/components/module-viewer/content/modals/EditHeadingModal.tsx

```diff
@@ -0,0 +1,136 @@
+import {
+  Select,
+  FormControl,
+  FormLabel,
+  Input,
+  Heading,
+} from "@chakra-ui/react";
+import React, { useContext, useState } from "react";
+import { HeadingBlockState } from "../../../../types/ContentBlockTypes";
+import { Modal } from "../../../common/Modal";
+import EditorContext from "../../../../contexts/ModuleEditorContext";
+import {
+  EditContentModalProps,
+  ValidHeadingSizes,
+} from "../../../../types/ModuleEditorTypes";
+
+const EditHeadingModal = ({
+  block,
+  index,
+  isOpen,
+  onClose,
+}: EditContentModalProps<HeadingBlockState>): React.ReactElement => {
+  const [text, setText] = useState(block.content.text ?? "why");
```

Comment:
"why"? Out of curiosity, could this ever happen? If not, can we make the field required so we don't have to do any nullish coalescing?

Reply:
I have no idea why i wrote this tbh, shouldn't need coalescing